server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -301,7 +333,7 @@ public SourceFieldMapper build() {
                 return enableRecoverySource ? LOGSDB_DEFAULT : LOGSDB_DEFAULT_NO_RECOVERY_SOURCE;
             }
         }
-        return enableRecoverySource ? DEFAULT : DEFAULT_NO_RECOVERY_SOURCE;
+        return getDefaultSourceMode(enableRecoverySource, INDEX_MAPPER_SOURCE_MODE_SETTING.get(c.getSettings()));
```

Comment:
I think in this case we do need to validate whether index.mapping.source.mode matches with what source mode is in the provided default instances?

Reply:
I fixed the issue...in the `isDefault` method we need to determine the mode first by evaluating the setting if exists or not. I will push the fix soon.